src/tests/ftest/soak/soak.py

```diff
@@ -0,0 +1,309 @@
+        self.soak_results = {}
+
+        self.partition = None
+
+        # initialize anything we rely on existing
+        self.pool = None
+        self.hostlist_servers = None
+
+        # get paths from the build_vars generated by build
+        with open('../../../.build_vars.json') as thefile:
+            build_paths = json.load(thefile)
+        self.basepath = os.path.normpath(build_paths['PREFIX']  + "/../")
+
+        # workdir was not successful, not sure why right now
+        self.tmpdir = self.basepath + "/install/tmp"
+        try:
+            os.makedirs(self.tmpdir)
+        except:
+            pass
+
+        # setup the DAOS python API
+        self.context = DaosContext(build_paths['PREFIX'] + '/lib/')
+
+        # start the servers
+        self.hostlist_servers = self.params.get("daos_servers",
+                                                '/run/hosts/test_machines/*')
+        filename = WriteHostFile.WriteHostFile(self.hostlist_servers,
+                                               self.workdir)
+        self.server_group = self.params.get("server_group", '/server/',
+                                            'daos_server')
+        print("Servers {} group {} basepath {}".format(self.hostlist_servers,
+                                                       self.server_group,
+                                                       self.basepath))
+        ServerUtils.runServer(filename, self.server_group, self.basepath)
+
+        # setup the storage
+        self.create_pool()
+
+    def tearDown(self):
+        ServerUtils.stopServer(hosts=self.hostlist_servers)
+
+    def test_soak_1(self):
+        """
+        Test ID: DAOS-2192
+        Test Description: This test runs 2 DAOS API IOR jobs.
+        :avocado: tags=soak1
+        """
+
+        try:
+            # turn job parameters into slurm script
+            script1 = self.build_ior_script('job1')
+
+            # queue it up to run and register a callback to retrieve results
+            job_id1 = slurm_utils.run_slurm_script(script1)
+            slurm_utils.register_for_job_results(job_id1, self, maxwait=3600)
+
+            # queue up a second job
+            script2 = self.build_ior_script('job2')
+            job_id2 = slurm_utils.run_slurm_script(script2)
+            slurm_utils.register_for_job_results(job_id2, self, maxwait=3600)            
+
+            # wait for all the jobs to finish
+            while len(self.soak_results) < 2:
+                time.sleep(10)
+
+            for job, result in self.soak_results.iteritems():
+                if result != "COMPLETED":
+                    self.fail("Soak job: {} didn't complete as expected: {}".
+                              format(job, result))
+
+        except (DaosApiError, IorUtils.IorFailed) as error:
+            self.fail("<Soak Test 1 Failed>\n {}".format(error))
+        finally:
+            try:
+                os.remove(script1)
+            except StandardError:
```

Comment:
Why StandardError? Is the idea to catch all Exceptions but not use the Exception keyword? I think something like OSError might fit better depending on the goal of these nested try blocks.